param/__init__.py

```diff
@@ -1534,6 +1534,125 @@ def _validate(self, val):
             self._length_bounds_check(self.rows, len(val), 'Row')
 
 
+class _XrBase:
+    """
+    Provides mixin methods for DataArray and Dataset. Not useful on its own.
+    """
+    __slots__ = []
+
+    def _validate_dims_coords(self, val):
+        if self.dims is not None:
+            self._validate_property(self.dims, val.dims, 'dimensions')
+
+        if self.coords is not None:
+            self._validate_property(self.coords, list(val.coords),
+                                           'coordinates')
+            if isinstance(self.coords, dict):
+                self._coord_check(self.coords, val.coords)
+
+
+    def _validate_property(self, expected, property, name):
+        difference = set(expected) - set([str(el) for el in property])
+        if difference:
+            msg = ('Provided DataArray {name} {found} do not contain '
```

Comment:
```suggestion
            msg = ('Provided DataArray {name} {found} does not contain '
```

Reply:
The argument I passed for `name` are all plural ("columns", "data variables", ...)